modules/command-engine/core/src/main/scala/surge/core/SurgePartitionRouter.scala

```diff
@@ -26,7 +26,17 @@ object SurgePartitionRouter {
       kafkaStreamsCommand: AggregateStateStoreKafkaStreams[JsValue],
       regionCreator: PersistentActorRegionCreator[String],
       signalBus: HealthSignalBusTrait,
-      isAkkaClusterEnabled: Boolean): SurgePartitionRouter = {
-    new SurgePartitionRouterImpl(config, system, partitionTracker, businessLogic, kafkaStreamsCommand, regionCreator, signalBus, isAkkaClusterEnabled)
+      isAkkaClusterEnabled: Boolean,
+      kafkaProducerOverride: Option[KafkaProducerTrait[String, Array[Byte]]] = None): SurgePartitionRouter = {
```

Comment:
I wasn't able to mock kafka producer for multi-jvm test. That's why this producer parameter has been introduced.